src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationWriterILGen.cs

```diff
@@ -2425,6 +2486,7 @@ internal void ILGenForCreateInstance(CodeGenerator ilg, Type type, Type cast, bo
             ilg.Call(Activator_CreateInstance);
             if (cast != null)
                 ilg.ConvertValue(Activator_CreateInstance.ReturnType, cast);
+            ilg.MarkLabel(labelReturn);
```

Comment:
In what situations will the code to find/use a parameterless constructor fail but Activator.CreateInstance will succeed?  If the type has a parameterless instance constructor, I'd expect the find/use code to find it, and if it doesn't, I'd expect Activator.CreateInstance to fail.  So I'm wondering if the Activator.CreateInstance code is now dead...?